src/billable-services/charge-items/charge-items-modal.component.tsx

```diff
@@ -0,0 +1,292 @@
+
+const DEFAULT_PAYMENT_OPTION = { paymentMode: '', price: 0 };
+
+const AddBillableStock: React.FC<{ editingItem?: any; onClose: () => void }> = ({ onClose, editingItem }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+
+  const [selectedItem, setSelectedItem] = useState<StockItem | null>(null);
+  const [searchTerm, setSearchTerm] = useState('');
+  const { serviceTypes, isLoading: isLoadingServicesTypes } = useServiceTypes();
+
+  const debouncedSearchTerm = useDebounce(searchTerm);
+
+  const { searchResults, error, isLoading } = useFetchChargeItems(debouncedSearchTerm);
+  const searchInputRef = useRef(null);
+  const handleSearchTermChange = (event: React.ChangeEvent<HTMLInputElement>) => setSearchTerm(event.target.value);
+  const { paymentModes, isLoading: isLoadingPaymentModes } = usePaymentModes();
+  const servicePriceSchema = z.object({
+    paymentMode: z.string().refine((value) => !!value, 'Payment method is required'),
+    price: z.union([
+      z.number().refine((value) => !!value, 'Price is required'),
+      z.string().refine((value) => !!value, 'Price is required'),
+    ]),
+  });
+  const paymentFormSchema = z.object({
+    payment: z.array(servicePriceSchema).min(1, 'At least one payment option is required'),
+  });
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      search: '',
+      payment: editingItem?.servicePrices || DEFAULT_PAYMENT_OPTION,
+      serviceType: editingItem?.serviceType,
+    },
+    resolver: zodResolver(paymentFormSchema),
+  });
+
+  const { fields, remove, append } = useFieldArray({ name: 'payment', control: control });
+
+  const handleAppendPaymentMode = useCallback(() => append(DEFAULT_PAYMENT_OPTION), [append]);
+  const handleRemovePaymentMode = useCallback((index) => remove(index), [remove]);
+
+  const handleChargeItemChange = useCallback((selectedItem: any) => {
+    setSelectedItem(selectedItem);
+  }, []);
+
+  const getPaymentErrorMessage = () => {
+    const paymentError = errors.payment;
+    if (paymentError && typeof paymentError.message === 'string') {
+      return paymentError.message;
+    }
+    return null;
+  };
+
+  const onSubmit = (data) => {
+    if (!selectedItem) {
+      showSnackbar({
+        title: t('missingItem', 'Missing item'),
+        subtitle: t('pleaseSelectAnItem', 'Please select a commodity before submitting'),
+        kind: 'error',
+      });
+      return;
+    }
+
+    // find selected paymentMode details
+    const servicePrices = data.payment.map((payment) => {
+      const mode = paymentModes.find((m) => m.uuid === payment.paymentMode);
+      return {
+        name: mode?.name || 'Unknown',
+        price: parseFloat(payment.price),
+        paymentMode: {
+          uuid: payment.paymentMode,
+          name: mode?.name || '',
+        },
+      };
+    });
+
+    const payload = {
+      name: selectedItem?.drugName,
```

Comment:
This name should be the name of the `payment type` **not** `drugName`

<img width="1007" height="221" alt="Screenshot 2025-07-30 at 16 10 07" src="https://github.com/user-attachments/assets/30b52fca-e767-439e-bcd2-3dcf35fa653c" />